### PR TITLE
feat: add power up and power down buttons

### LINKED ISSUE #1333: Power up and power down buttons

### PR DESCRIPTION
**Pull request by Sekhmet**
Fixes #1333 

### Changes

* add power up and power down buttons
* add PowerUpOrDown modal

### Test plan

* [x] Go to your profile.
* [x] Click on Power up button and try it.
* [x] Click on Power down button and try it.

### Demo

image

image

### CODE REVIEW

**Comment by bonustrack** [comment]
Great work on that! Can we display the value in STEEM but not in VESTS? VESTS should not be visible for users.

**Comment by bonustrack** [comment]
@Sekhmet Can we have always 6 decimals on the VEST amount when we sending user to power down/up?

File: src/client/wallet/PowerUpOrDown.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      amount: value,
+    });
+  };
+
+  handleContinueClick = () => {
+    const { form, user, down, totalVestingShares, totalVestingFundSteem } = this.props;
+    form.validateFields({ force: true }, (errors, values) => {
+      const vests = (
+        values.amount / formatter.vestToSteem(1, totalVestingShares, totalVestingFundSteem)
+      ).toFixed(6);
```
Comment (jm90m) [inline src/client/wallet/PowerUpOrDown.js]: we're fixing vests to 6 decimal places?

Comment (Sekhmet) [reply src/client/wallet/PowerUpOrDown.js]: Otherwise transaction would be invalid:
`decimals == 6: Incorrect decimal places`

**Comment by Sekhmet** [comment]
How should it work then? I used URLs from the original issue.
`transfer-to-vesting` users STEEM and `withdraw-vesting` uses VESTS?

**Comment by bonustrack** [comment]
Yes power-up with STEEM and power-down VESTS

**Pull request merged**